### PR TITLE
Next

### LINKED ISSUE #1274: Remove gatsby-plugin-react-helmet
## Remove dependency gatsby-plugin-react-helmet

Support for the Gatsby Head API was added in gatsby@4.19.0, and `gatsby-plugin-react-helmet` is no longer needed.

## Detailed description

```shell
warn gatsby-plugin-react-helmet: Gatsby now has built-in support for modifying the document head. Learn more at https://gatsby.dev/gatsby-head
```

### PR DESCRIPTION
**Pull request by eng618**
Rel #1324 
Closes #1274 

This is a major Gatsby upgrade to v5.

#### Changelog

**New**

- Gatsby v5
- Upgrade node to v18
- Upgrades to latest mdx
- Replace `gatsby-plugin-lunr` with `gatsby-plugin-local-search`
- Migration documentation

**Changed**

- Updated layouts

**Removed**

- gatsby-plugin-react-helmet

### CODE REVIEW

File: packages/gatsby-theme-carbon/package.json [hunk 1]
```diff
@@ -12,55 +12,56 @@
...
-    "gatsby-plugin-offline": "^5.25.0",
-    "gatsby-plugin-sass": "5.25.0",
-    "gatsby-plugin-sass-resources": "^3.0.1",
-    "gatsby-plugin-sharp": "^4.25.1",
-    "gatsby-remark-copy-linked-files": "^5.25.0",
-    "gatsby-remark-images": "^6.25.0",
-    "gatsby-remark-responsive-iframe": "^5.25.0",
-    "gatsby-remark-smartypants": "^5.25.0",
+    "gatsby-plugin-manifest": "^5.12.0",
+    "gatsby-plugin-mdx": "^5.12.0",
```
Comment (muditjuneja) [inline packages/gatsby-theme-carbon/package.json:45]: Hi @eng618 I see you have already updated the version here and still we are using defaultLayouts in the config file. Can you help me with some context around what this version upgrade has brought/deprecated for which we need to make changes?

File: packages/example/src/pages/guides/v4-migration.mdx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+   ```
+
+1. Update graphql queries using the codemod:
+
+   ```shell
+   npx gatsby-codemods@latest sort-and-aggr-graphql
+   ```
+
+1. Homepage and default layouts have been moved from **templates** to **layout**
+   components. Update any shadowing or links to template layouts.
```
Comment (alisonjoseph) [inline packages/example/src/pages/guides/v4-migration.mdx:62]: Should we move this under breaking changes?

```jsx
// Before
import Layout from 'gatsby-theme-carbon/src/templates/Default';
// After
import Layout from 'gatsby-theme-carbon/src/components/Layouts/Default';
```

If you are shadowing the homepage you need to update the file path
`packages/example/src/components/Homepage.js`

```jsx
// Before
import HomepageTemplate from 'gatsby-theme-carbon/src/templates/Homepage';
// After
import HomepageTemplate from 'gatsby-theme-carbon/src/components/Layouts/Homepage';
```

File: packages/example/src/pages/guides/v4-migration.mdx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+</PageDescription>
+
+## Breaking changes
+
+1. Bumped minimum version of Gatsby to `v5`
+1. Gatsby files (i.e. gatsby-config.js and gatsby-node.js), need to be converted
+   to **.mjs** (node modules) files.
+1. TBD...
+
```
Comment (alisonjoseph) [inline packages/example/src/pages/guides/v4-migration.mdx:24]: Copying over some "breaking" things I noticed when migrating Carbon.

1. Markdown link with title needs double quotes
```jsx
// Before
[I'm a markdown link with title](https://www.carbondesignsystem.com "Carbon's Homepage")
// After
[I'm a markdown link with title](https://www.carbondesignsystem.com 'Carbon's Homepage')
```

1. HTML-style comments will no longer work inside of mdx files, you will need to use JavaScript-style comments with curly braces:
```jsx
// Before
<!-- Place comments here -->
// After
{/* Place comments here */}
```

File: packages/gatsby-theme-carbon/gatsby-config.mjs [hunk 4]
```diff
@@ -112,12 +123,14 @@ module.exports = (themeOptions) => {
...
-            home: require.resolve('./src/templates/Homepage.js'),
-            ...extraLayouts,
+          mdxOptions: {
+            remarkPlugins: [remarkGfm, ...remarkPlugins],
+            rehypePlugins: [rehypeAddCodeMetaData],
           },
+          // defaultLayouts: {
+          //   default: require.resolve('./src/templates/Default.js'),
+          //   home: require.resolve('./src/templates/Homepage.js'),
+          // },
```
Comment (alisonjoseph) [inline packages/gatsby-theme-carbon/gatsby-config.mjs:133]:
```suggestion
```

Can we remove this?

**Review by alisonjoseph — COMMENTED** [review]
Added a couple comments around migration

**Review by alisonjoseph — APPROVED** [review]

**Pull request merged**